src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    if (this.length === 0) {
+      return '';
+    }
```

Comment:
```suggestion
```
This check is redundant. If `this.length === 0` the loop won't even start:
`for (let i = 0; i < this.length - 1; i++)` ===> `for (let i = 0; i < 0 - 1; i++)`
here 0 will never be less than -1, so there will be no single iteration in this loop